gen/aws/templates/advanced/advanced-master.json

```diff
@@ -46,6 +46,12 @@
       "Type": "String",
       "Default": "m3.xlarge",
       "Description" : "\nRegion-specific instance type. E.g. m3.xlarge"
+    },
+    "NumMasters" : {
```

Comment:
A bunch of internal code code which is emitted into the cloud-init for the masters hard-sets values derived from the number of masters, so you can't do one advanced master template that has a variable number of masters.

That said, having all the different advanced-master templates, and having zen be able to select which to use based on a number is feasible.